packages/http/src/metadata.ts

```diff
@@ -135,6 +143,26 @@ export function getRequestVisibility(verb: HttpVerb): Visibility {
   }
 }
 
+/**
+ * Determines the visibility to use for a request with the given verb, either by evaluating
+ * the defaults or applying the results of `@requestVisibility`. This should probably be used
+ * in place of `getRequestVisibility` or `getOperationRequestVisibility` in most cases.
+ *
+ * @param operation The HTTP operation to evaluate.
+ * @param verb The HTTP verb for the operation.
+ */
+export function resolveRequestVisibility(
```

Comment:
This becomes the quintessential method you would want to call. I'm not sure if there is a good way to dissuade people from calling `getRequestVisibility` and `getOperationRequestVisibility` directly.

Reply:
Again, haven't looked at everything, but we could not export getOperationRequestVisibility or mark it `/** @internal */` since it hasn't existed before. `getRequestVisibility(verb)` could be marked `/** @deprecated Use resolveRequestVisibility */`. If we need to use it under the covers, maybe move the body to an internal function with a new name like getRequestVisbilityForVerb() and have the deprecated function with the old name be a stub that calls that.

Reply:
Well, I exported it because when we have a decorator we usually export the "getDecorator" variant... I just couldn't use the same name because we already export something with the name.

But realistically, in most cases I think you would want the logic of `resolveRequestVisibility` which essentially checks the `@requestVisibility` decorator first and falls back to the verb if needed.

Reply:
Ah, I see. I think it would be OK to not export the 1:1 decorator getter since there's a good chance of using it instead of the thing that takes everything else into account. We could always export it later if it turns out that it's required outside the library for something, which I doubt will happen. I also don't feel too strongly on any of this, and it would be fine to me, for example, to just add to the doc comments of the other functions to say `you probably want {@link resolveRequestVisibility}`